proto/requests.proto

```diff
@@ -135,6 +135,9 @@ message GetAccountStateDeltaRequest {
 message GetAccountProofsRequest {
     // List of account IDs to get states.
     repeated account.AccountId account_ids = 1;
+    // Account code commitments corresponding to the last-known `AccountCode` for requested 
+    // accounts. Responses will include only the ones that are not known to the caller.
+    repeated digest.Digest code_commitments = 2;
```

Comment:
I would probably mention that the number of code commitments must be exactly the same as the number of account IDs.

Also, I think we should return the code only if `include_headers` was set to true.

Reply:
I made this change, but I'm not sure it improves the API too much. Basically I was seeing this parameters as a set of known commitments that the node could check against without it necessarily being the same length as the requested accounts (maybe the caller knows 1 out of 3 requested accounts and so sending 1 commitment would be fine). Although having the check is nice to not receive an unnecessarily-long parameter.

Reply:
But how would we be able to map which account ID corresponds to which commitment? We could refactor the request to use `(AccountID, CodeRoot)` tuples - but I think this may be complicating it even more.

One thing I don't love about the current approach is that now the user has to provide a digest for every account ID (even if they are not interested in account code). But I'm not seeing an easy way around this.

Reply:
Ah - I get it now: you were thinking of the list of code commitments as not mapped to a specific ID but for a request in general. I think your view here is better. But I think we should still check to make sure the number of commitments is not greater than the number of IDs.

Reply:
Do we need to correlate request account IDs with request code commitments? The response account headers contain the code root already, so the response for a single account ID would either:

- Contain `None` for `account_code`, but the `AccountHeader` would have a `code_commitment` that the user knows (ie, which was on the request)
- Contain `Some(code)`, where the `AccountHeader` would have a `code_commitment` the user does not know (ie, it was not on the request's list of commitments)

Doesn't really matter for which account we thought the code commitment corresponded to on the request, because with the response we should have all the pieces.
So if `account_1` and `account_2` had the same code commitment (according to the API user's perspective), the list they would send would only contain one code commitment, but maybe the response comes back with each account having a new `AccountCode` which could even be different from each other. 
But very possible I'm missing something basic here.